src/main/java/com/daitangroup/api/controller/PersonController.java

```diff
@@ -1,35 +1,124 @@
 package com.daitangroup.api.controller;
 
+import com.daitangroup.api.exception.CpfAlreadyRegisteredException;
 import com.daitangroup.api.model.Person;
-import com.daitangroup.api.repository.PersonRepository;
+import com.daitangroup.api.model.PersonReturn;
+import com.daitangroup.api.services.GarageService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
-
 import java.util.Optional;
 
 @RestController
-@RequestMapping("/person")
+@RequestMapping("/persons")
 public class PersonController {
 
     @Autowired
-    private PersonRepository repository;
+    private GarageService garageService;
 
     @RequestMapping(method = RequestMethod.GET, value = "/{id}")
-    @ResponseBody
-    public ResponseEntity<Person> find(@PathVariable Long id) {
+    public ResponseEntity<Person> findPersonById(@PathVariable Long id) {
+
+        Optional<Person> personResult = garageService.findPersonById(id);
+
+        if (personResult.isPresent()) {
+            return new ResponseEntity<>(personResult.get(), HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<Iterable> findAllPersons() {
+
+        Iterable<Person> personsResult = garageService.findAllPersons();
+
+        if (personsResult.iterator().hasNext()) {
+            return new ResponseEntity<>(personsResult, HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
If is ok to return an empty list in case of none found. Otherwise, change the /vehicles endpoint to have the same behavior as this.

Reply:
Also, make this the first method of the class

Reply:
Did you mean that i need to pass a new empty list on the response entity?
What did you mean with: "_...change the /vehicles endpoint to have the same behavior as this._" sorry i didn't
understood.

And why this should be the first method? Just curiosity.

Reply:
> Did you mean that i need to pass a new empty list on the response entity?
> What did you mean with: "...change the /vehicles endpoint to have the same behavior as this." > sorry i didn't
> understood.

For consistency, your API should behave the same way across all endpoints, unless there is an explicit business rule saying otherwise. So, for all the endpoint returning a collection, when no element found, either make all of the endpoints return an empty collection, or make them all return 404. Just be sure to make all of them to behave the same way. 

> And why this should be the first method? Just curiosity.

This method is the base endpoint for this controller, so is easier to spot it if it is the first one. As it has no value in the annotation, is difficult to guess and find which method is the base endpoint.